### PR TITLE
introduce new doc structure to sci-viz

### PR DESCRIPTION
**Pull request by A-Baji**
https://a-baji.github.io/sci-viz

### CODE REVIEW

File: docs/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    user: ${HOST_UID}:anaconda
+    ports:
+      - 80:80
+    command:
+      - sh
+      - -c
+      - |
+        git config --global --add safe.directory /main
+        set -e
+        export PATCH_VERSION=$$(cat /main/package.json | jq -r '.version')
```
Comment (ana530) [inline docs/docker-compose.yaml:31]:
```suggestion
```

File: docs/src/concepts/grids.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Grids
+
+Sci-Viz produces custom visualizations by putting `grids` on `pages` and then filling them with visualization `components`. Currently there are two types of grids **Fixed** and **Dynamic**
+
+## Fixed mode grid
```
Comment (ana530) [inline docs/src/concepts/grids.md:5]:
```suggestion
## Fixed grid
```

File: docs/src/concepts/grids.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+A fixed mode grid requires all components to explicitly give their position and size on the grid.
```
Comment (ana530) [inline docs/src/concepts/grids.md:7]:
```suggestion
A fixed grid requires all components to explicitly give their position and size on the grid.
```

File: docs/src/concepts/grids.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+A fixed `grid` takes four arguments:
+
+- `type:` indicates the type of grid, in this case `type: fixed`
+- `columns:` the number of columns that the grid will have
+- `row_height:` the height of each row in pixels
+- `components:` a yaml dictionary of components to be spawned in the grid
+
+## Dynamic grid mode
```
Comment (ana530) [inline docs/src/concepts/grids.md:16]:
```suggestion
## Dynamic grid
```

File: docs/src/concepts/dev_instructions.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Developer Instructions
+
+There are a couple issues to address if you are collaborating on this project
+
+- devs will have have to point the submodule to their own fork of pharus if they need to edit pharus to support new features for sci-viz.
```
Comment (ana530) [inline docs/src/concepts/dev_instructions.md:5]:
```suggestion
- Contributors will have have to point the submodule to their own fork of pharus if they need to edit pharus to support new features for sci-viz.
```

File: docs/src/concepts/references.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# References
+
+- [https://datajoint.org](https://datajoint.org) -- start page
+- [https://github.com/datajoint/pharus](https://github.com/datajoint/pharus) -- Pharus (a DataJoint REST API backend)
+- [https://datajoint.com/docs/](https://datajoint.com/docs/) -- up-to-date documentation
+- [https://tutorials.datajoint.org](https://tutorials.datajoint.org) -- step-by-step tutorials
```
Comment (ana530) [inline docs/src/concepts/references.md:6]: delete file because links are outdated or unneeded
```suggestion
```

File: docs/mkdocs.yaml [hunk 7]
```diff
@@ -0,0 +1,154 @@
+# ---------------------- PROJECT SPECIFIC ---------------------------
+
+site_name: Sci-Viz
```
Comment (ana530) [inline docs/mkdocs.yaml:3]:
```suggestion
site_name: SciViz
```

File: docs/mkdocs.yaml [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+site_url: http://localhost/docs/sci-viz
```
Comment (ana530) [inline docs/mkdocs.yaml:4]:
```suggestion
site_url: http://datajoint.com/docs/sci-viz
```

File: docs/mkdocs.yaml [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+repo_url: https://github.com/datajoint/sci-viz
+repo_name: datajoint/sci-viz
+nav:
+  - Getting Started: getting_started.md
+  - Concepts:
+      - Dynamic Spec Sheet: concepts/spec_sheet.md
+      - Grids: concepts/grids.md
+      - Components: concepts/components.md
+      - Developer Instructions: concepts/dev_instructions.md
+      - References: concepts/references.md
```
Comment (ana530) [inline docs/mkdocs.yaml:14]:
```suggestion
```

File: docs/mkdocs.yaml [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+          format: !!python/name:pymdownx.superfences.fence_code_format
+  - pymdownx.tabbed:
+      alternate_style: true
+  - pymdownx.highlight:
+      linenums: true
+  - pymdownx.inlinehilite
+  - pymdownx.snippets
+
+extra:
+  PATCH_VERSION: !ENV PATCH_VERSION
```
Comment (ana530) [inline docs/mkdocs.yaml:114]: please look for patch versions to omit
```suggestion
```

**Review by ana530 — CHANGES_REQUESTED** [review]
some changes to be made!

File: docs/src/.overrides/.icons/main/company-logo.svg [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+<!-- Generator: Adobe Illustrator 15.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 width="260.479px" height="365.463px" viewBox="0 0 260.479 365.463" enable-background="new 0 0 260.479 365.463"
+	 xml:space="preserve">
+<g>
+	<path d="M4.638,361.434c0,0,124.071-124.072,135.896-357.434c0,0,29.555,189.053,115.266,256.988
+		C255.799,260.988,173.038,249.174,4.638,361.434z"/>
+</g>
+</svg>
```
[the remark this answers is not in the document]
Comment (A-Baji) [reply docs/src/.overrides/.icons/main/company-logo.svg:11]: This file actually is used. woops!

File: .github/workflows/development.yaml [hunk 12]
```diff
@@ -39,6 +39,24 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Deploy docs
+        env:
+          GOOGLE_ANALYTICS_KEY: ${{secrets.GOOGLE_ANALYTICS_KEY}}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          export MODE=PUSH
+          export PACKAGE=sci-viz
+          export UPSTREAM_REPO=https://action-user:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
```
[the remark this answers is not in the document]
Comment (A-Baji) [reply .github/workflows/development.yaml:56]: This change breaks the deployment workflow

**Review by ana530 — APPROVED** [review]

**Review by ana530 — APPROVED** [review]

**Review by ana530 — APPROVED** [review]

**Review by ana530 — APPROVED** [review]

**Pull request merged**